src/main/java/frc/team2412/robot/commands/shooter/ShooterTargetCommand.java

```diff
@@ -6,58 +6,39 @@
 
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import frc.team2412.robot.subsystem.ShooterSubsystem;
-import frc.team2412.robot.subsystem.ShooterVisionSubsystem;
+import frc.team2412.robot.subsystem.TargetLocalizer;
 import frc.team2412.robot.util.ShooterDataDistancePoint;
 
 public class ShooterTargetCommand extends CommandBase {
     private final ShooterSubsystem shooter;
-    private final ShooterVisionSubsystem vision;
+    private final TargetLocalizer localizer;
     private final BooleanSupplier turretEnable;
 
-    public ShooterTargetCommand(ShooterSubsystem shooter, ShooterVisionSubsystem vision) {
-        this(shooter, vision, () -> false);
+    public ShooterTargetCommand(ShooterSubsystem shooter, TargetLocalizer localizer) {
+        this(shooter, localizer, () -> false);
     }
 
-    public ShooterTargetCommand(ShooterSubsystem shooter, ShooterVisionSubsystem vision, BooleanSupplier turretEnable) {
+    public ShooterTargetCommand(ShooterSubsystem shooter, TargetLocalizer localizer, BooleanSupplier turretButton) {
         this.shooter = shooter;
-        this.vision = vision;
-        this.turretEnable = turretEnable;
+        this.localizer = localizer;
+        turretEnable = turretButton;
         addRequirements(shooter);
     }
 
     @Override
     public void execute() {
-        if (!shooter.enableTurret) {
-            return;
-        }
-
-        double distance = 0, yaw = 0;
-
-        if (vision.hasTarget()) {
-            distance = vision.getDistance() + shooter.getDistanceBias();
-            yaw = vision.getYaw() + shooter.getTurretAngleBias();
-        }
 
         if (ShooterConstants.dataPoints != null) {
-            ShooterDataDistancePoint shooterData = ShooterConstants.dataPoints.getInterpolated(distance);
+            ShooterDataDistancePoint shooterData = ShooterConstants.dataPoints
+                    .getInterpolated(localizer.getAdjustedDistance());
             shooter.setHoodAngle(shooterData.getAngle());
             shooter.setFlywheelRPM(shooterData.getRPM());
         }
-
         if (turretEnable.getAsBoolean()) {
-            shooter.updateTurretAngle(yaw);
+            shooter.setTurretAngle(shooter.getTurretAngle() + localizer.getYaw() + localizer.yawAdjustment());
```

Comment:
It'd be cleaner to use `updateTurretAngle(localizer.getYaw() + localizer.yawAdjustment())` (or, even cleaner, add a `getAdjustedYaw()` to mirror `getAdjustedDistance()` and use `updateTurretAngle(localizer.getAdjustedYaw())`), but it doesn't matter *too* much.

Reply:
I second this, would be prettier and easier to read